banyan-macos-v3

```diff
@@ -0,0 +1,127 @@
+echo "Installing using deploy key: $DEPLOYMENT_KEY"
+echo "Installing app version: $APP_VERSION"
+
+logged_on_user=$( echo "show State:/Users/ConsoleUser" | scutil | awk '/Name :/ && ! /loginwindow/ { print $3 }' )
+echo "Installing app for user: $logged_on_user"
+
+global_config_dir="/private/etc/banyanapp"
+tmp_dir="/tmp"
+
+
+function create_config() {
+	echo "Creating mdm-config json file"
+
+	global_config_file="${global_config_dir}/mdm-config.json"
+
+	deploy_user=""
+	deploy_email=""
+
+	# contact Banyan Support to enable the feature that will allow you to issue 
+	# a device certificate for a specific user instead of the default **STAGED USER**
+	#
+    # you can get user and email via a custom plist file deployed via Device Manager
+	# or try one of these techniques: https://github.com/pbowden-msft/SignInHelper
+	#if [[ -e "/Library/Managed Preferences/custom.plist" ]]; then
+	#	deploy_user=$( defaults read "/Library/Managed Preferences/custom.plist" name )
+	#	deploy_email=$( defaults read "/Library/Managed Preferences/custom.plist" email )	
+	#fi
+
+	# the config below WILL NOT install your org's Banyan Private Root CA
+	# on MacOS, we recommend using your Device Manager to push down the Private Root CA
+
+	mdm_config_json='{
+		"mdm_invite_code": "REPLACE_WITH_INVITE_CODE",
+		"mdm_deploy_user": "REPLACE_WITH_USER",
+		"mdm_deploy_email": "REPLACE_WITH_EMAIL",
+		"mdm_device_ownership": "C",
+		"mdm_ca_certs_preinstalled": true,
+		"mdm_skip_cert_suppression": false,
+		"mdm_vendor_name": "JAMF",
+		"mdm_start_at_boot": true,
+		"mdm_hide_on_start": true	
+	}'
+
+	mkdir -p "$global_config_dir"
+	echo "$mdm_config_json" > "${global_config_file}"
+	sed -i '' "s/REPLACE_WITH_INVITE_CODE/${INVITE_CODE}/" "${global_config_file}"
+	sed -i '' "s/REPLACE_WITH_USER/${deploy_user}/" "${global_config_file}"
+	sed -i '' "s/REPLACE_WITH_EMAIL/${deploy_email}/" "${global_config_file}"
+}
+
+
+function download_install() {
+	echo "Downloading installer PKG"
+
+	full_version="${APP_VERSION}"
+	dl_file="${tmp_dir}/Banyan-${full_version}.pkg"
+
+	if [[ -f "${dl_file}" ]]; then
+		echo "Installer PKG already downloaded"
+	else
+		curl -sL "https://www.banyanops.com/app/releases/Banyan-${full_version}.pkg" -o "${dl_file}"
+	fi
+
+    #Install PKG
+    sudo installer -pkg "${dl_file}" -target /
+    sleep 3
+}
+
+function stage() {
+	echo "Running staged deployment"
+    /Applications/Banyan.app/Contents/Resources/bin/banyanapp-admin stage --key=$DEPLOYMENT_KEY
+    sleep 3
+	echo "Staged deployment done. Start the app to complete registration."
+}
+
+
+function start_app() {
+	echo "Starting the Banyan app as: $logged_on_user"
+	sudo -H -u "${logged_on_user}" open /Applications/Banyan.app
+	sleep 5
+}
+
+
+function stop_app() {
+	echo "Stopping Banyan app"
+	killall Banyan
+	sleep 2
+}
+
+
+if [[ "$INVITE_CODE" = "upgrade" && "$DEPLOYMENT_KEY" = "upgrade" ]]; then
+	echo "Running upgrade flow"
+	stop_app
+	download_install
+	start_app
```

Comment:
`start_app` can be moved outside of the `if` statement since both conditions call it.